### ISSUE TITLE
Use `base-compat` for backwards compatibility

### ISSUE THREAD

**Issue by wenkokke** [vehicle-lang/vehicle#311]
E.g. for `List.singleton` import [`Data.List.Compat`](https://hackage.haskell.org/package/base-compat-0.12.2/docs/Data-List-Compat.html) as `List`.

**Comment by MatthewDaggitt** [comment]
I don't think this is either necessary or planned. We seem to only have relatively minimal problems with compatibility with prior Haskell versions so closing.
